pyccl/tracers.py

```diff
@@ -262,6 +321,95 @@ def get_bessel_derivative(self):
             return []
 
         return np.array([t.der_bessel for t in self._trc])
+	
+    def MG_add_tracer(self, cosmo, kernel, mg_transfer, 
+                    k_MG=None, der_bessel=0, der_angles=0,
+                    bias_transfer_a=None, bias_transfer_k=None):
+        """ function to set mg_transfer in the right format and add MG tracers 
+            for different cases including different cases and biases like intrinsic 
+            alignements (IA) when present
+        """
+        # case with no astro biases
+        if ((bias_transfer_a is None) and (bias_transfer_k is None)):    
+            if (isinstance(k_MG, (list, np.ndarray))): 		
```

Comment:
I don't really understand why you are splitting this into three if statements - looks like each case does the same thing? Am I missing something?

Reply:
Oh, I see the first one has a transfer_ka while the latter two have transfer_a - but the second two still seem to be the same?

Reply:
Good point. In principle, that is right and I could condense the two last into one. The reason I did not is that for future development in case some further needs to treat these cases differently. I added a comment to explain this but I can do it if you feel it is best.

Reply:
Building in flexibility makes sense to me. Happy for you to leave it.